### PR TITLE
doc/mgr/orchestrator: Add Cephfs

### PR DESCRIPTION
**Pull request by sebastian-philipp**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/mgr/orchestrator.rst [hunk 1]
```diff
@@ -256,9 +220,29 @@ the id is the numeric OSD ID, for MDS services it is the file system name::
     ceph orch daemon status <type> <instance-name> [--refresh]
 
 
+.. _orchestrator-cli-cephfs:
+    
+Depoying CephFS
+===============
+
+In order to set up a :term:`CephFS`, simply execute::
```
Comment (asettle) [inline doc/mgr/orchestrator.rst]:
```suggestion
In order to set up a :term:`CephFS`, execute::

File: doc/mgr/orchestrator.rst [hunk 2]
```diff
@@ -256,9 +220,29 @@ the id is the numeric OSD ID, for MDS services it is the file system name::
...
+
+    ceph fs volume create <fs_name> <placement spec>
+    
+Where
```
Comment (asettle) [inline doc/mgr/orchestrator.rst]:
```suggestion
Where:

Comment (asettle) [reply doc/mgr/orchestrator.rst]: You could also do "Where..." but I'm not even sure this needs to be two separate dot points. I'd actually just recommend that you put it all together.

`Where ``name`` is the name of the CephFS, ``placement`` is a :ref:`orchestrator-cli-placement-spec`.`

File: doc/mgr/orchestrator.rst [hunk 3]
```diff
@@ -256,9 +220,29 @@ the id is the numeric OSD ID, for MDS services it is the file system name::
...
+
+* ``name`` is the name of the CephFS
+* ``placement`` is a :ref:`orchestrator-cli-placement-spec`.
+    
+This command will 
+
+* create the required Ceph pools, 
+* create the new CephFS
+* deploy mds servers.
```
Comment (asettle) [inline doc/mgr/orchestrator.rst]: Ditto with this one.

File: doc/mgr/orchestrator.rst [hunk 4]
```diff
@@ -256,9 +220,29 @@ the id is the numeric OSD ID, for MDS services it is the file system name::
...
+In order to set up a :term:`CephFS`, simply execute::
+
+    ceph fs volume create <fs_name> <placement spec>
+    
+Where
+
+* ``name`` is the name of the CephFS
+* ``placement`` is a :ref:`orchestrator-cli-placement-spec`.
+    
+This command will
```
Comment (asettle) [inline doc/mgr/orchestrator.rst]:
```suggestion
This command will:

File: doc/mgr/orchestrator.rst (hunk 3 above)
Comment (asettle) [reply doc/mgr/orchestrator.rst]: `This command will create the required Ceph pools, create the new CephFS, and deploy mds servers.`

There's no reason to have it split up.

File: doc/mgr/orchestrator.rst [hunk 5]
```diff
@@ -285,7 +269,46 @@ Start/stop/reload::
...
     ceph orch daemon {start,stop,reload} <type> <daemon-id>
+    
+
+.. _orchestrator-cli-placement-spec:
+    
+Placement Specification
+=======================
+
+In order to allow the orchestrator to deploy a *service*, it needs to
+know how many and where it should deploy *daemons*. The orchestrator therefore
```
Comment (asettle) [inline doc/mgr/orchestrator.rst]:
```suggestion
know how many and where it should deploy *daemons*. The orchestrator

File: doc/mgr/orchestrator.rst [hunk 6]
```diff
@@ -285,7 +269,46 @@ Start/stop/reload::
...
+
+* To deploy three *daemons*, simply specify the count: ``3``
+* To deploy *daemons* on specific hosts, specify the host names: ``host1 host2 host3``
+* To deploy *daemons* on a subset of hosts, also specify the count: ``2 host1 host2 host3``
+
+    
+Configuring the Orchestrator CLI
+================================
+
+To enable the orchestrator, please select the orchestrator module to use
```
Comment (asettle) [inline doc/mgr/orchestrator.rst]:
```suggestion
To enable the orchestrator, select the orchestrator module to use

File: doc/mgr/orchestrator.rst [hunk 7]
```diff
@@ -285,7 +269,46 @@ Start/stop/reload::
...
+with the ``set backend`` command::
+
+    ceph orch set backend <module>
+
+For example, to enable the Rook orchestrator module and use it with the CLI::
 
+    ceph mgr module enable rook
+    ceph orch set backend rook
+
+You can then check backend is properly configured::
```
Comment (asettle) [inline doc/mgr/orchestrator.rst]:
```suggestion
Check the backend is properly configured::

File: doc/mgr/orchestrator.rst [hunk 8]
```diff
@@ -285,7 +269,46 @@ Start/stop/reload::
...
+
+    ceph orch status
+
+Disable the Orchestrator
+------------------------
+
+To disable the orchestrator again, use the empty string ``""``::
```
Comment (asettle) [inline doc/mgr/orchestrator.rst]:
```suggestion
To disable the orchestrator, use the empty string ``""``::

File: doc/mgr/orchestrator_modules.rst [hunk 9]
```diff
@@ -237,7 +224,25 @@ Devices
...
+.. orchestrator-modules-placement::
+
+Placement
+---------
+
+A :ref:`orchestrator-cli-placement-spec` defines the placement of
+daemons of a specifc service.
+
+In general, stateless services do not require any specific placement
+rules, as they can run anywhere that sufficient system resources
```
Comment (asettle) [inline doc/mgr/orchestrator_modules.rst]:
```suggestion
rules as they can run anywhere that sufficient system resources

File: doc/mgr/orchestrator_modules.rst [hunk 10]
```diff
@@ -237,7 +224,25 @@ Devices
...
+are available.  However, some orchestrators may not include the
```
Comment (asettle) [inline doc/mgr/orchestrator_modules.rst]:
```suggestion
are available. However, some orchestrators may not include the

File: doc/mgr/orchestrator_modules.rst [hunk 11]
```diff
@@ -237,7 +224,25 @@ Devices
...
+functionality to choose a location in this way, so we can optionally
```
Comment (asettle) [inline doc/mgr/orchestrator_modules.rst]:
```suggestion
functionality to choose a location in this way. Optionally, you can

**Comment by sebastian-philipp** [comment]
updated.

**Comment by sebastian-philipp** [comment]
jenkins render docs

File: doc/mgr/orchestrator.rst [hunk 12]
```diff
@@ -285,7 +264,46 @@ Start/stop/reload::
...
+In order to allow the orchestrator to deploy a *service*, it needs to
+know how many and where it should deploy *daemons*. The orchestrator 
+defines a placement specification:
+
+* To deploy three *daemons*, simply specify the count: ``3``
+* To deploy *daemons* on specific hosts, specify the host names: ``host1 host2 host3``
+* To deploy *daemons* on a subset of hosts, also specify the count: ``2 host1 host2 host3``
+
+    
+Configuring the Orchestrator CLI
```
Comment (liewegas) [inline doc/mgr/orchestrator.rst]: drop the `` CLI`` suffix

File: doc/mgr/orchestrator.rst [hunk 13]
```diff
@@ -285,7 +264,46 @@ Start/stop/reload::
...
+
+To enable the orchestrator, select the orchestrator module to use
+with the ``set backend`` command::
+
+    ceph orch set backend <module>
+
+For example, to enable the Rook orchestrator module and use it with the CLI::
 
+    ceph mgr module enable rook
+    ceph orch set backend rook
```
Comment (liewegas) [inline doc/mgr/orchestrator.rst]: We might as well show how to enable cephadm here too, since there are only 2 backends?

Comment (sebastian-philipp) [reply doc/mgr/orchestrator.rst]: I hope no user ever will need this.

File: doc/mgr/orchestrator.rst (hunk 12 above)
Comment (sebastian-philipp) [reply doc/mgr/orchestrator.rst]: Thing is: Either you use Rook, then you don't need to set up the orchestrator, or you use cephadm then you also don't need to set up the orchestrator. I was close before moving this section to the developer documentation.

Comment (liewegas) [reply doc/mgr/orchestrator.rst]: ah, true!

**Pull request merged**